episodes/09-open-and-plot-vector-layers.Rmd

```diff
@@ -0,0 +1,195 @@
+---
+title: "Open and Plot Vector Layers"
+teaching: 25
+exercises: 5
+---
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- How can you read, examine and visualize point, line and polygon vector data in R?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+After completing this episode, participants should be able to…
+
+- Differentiate between point, line, and polygon vector data.
+- Load vector data into R.
+- Access the attributes of a vector object in R.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor
+
+Make sure that the `sf` package and its dependencies are installed before the 
+workshop. The installation can be lengthy, so allocate enough extra time 
+before the workshop for solving installation problems. We recommend one
+or two installation 'walk-in' hours on a day before the workshop. Also, 15-30 
+minutes at the beginning of the first workshop day should be enough to tackle 
+last-minute installation issues.
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+::: prereq
+
+If you have not installed the `sf` package yet, run `install.packages("sf")` first. Note that the `sf` package has some external dependencies, namely GEOS, PROJ.4, GDAL and UDUNITS, which need to be installed beforehand. Follow the workshop [setup instructions](../learners/setup.md) for the installation of `sf` and its dependencies.
+
+:::
+
+First we need to load the packages we will use in this lesson. We will use the package `tidyverse` with which you are already familiar from the previous lesson. In addition, we need to load the [`sf`](https://r-spatial.github.io/sf/) package for working with spatial vector data. 
+
+```{r lib, message=FALSE}
+library(tidyverse)  # wrangle, reshape and visualize data
+library(sf)         # work with spatial vector data
+```
+
+::: callout
+
+# The `sf` package
+
+`sf` stands for Simple Features which is a standard defined by the Open Geospatial Consortium for storing and accessing geospatial vector data. PostGIS, for instance, uses the same standard, so PostGIS users might recognise the functions used by the `sf` package. 
+
+:::
+
+## Import shapefiles
+
+Let's start by opening a shapefile. Shapefiles are a common file format to store spatial vector data used in GIS software. We will read a shapefile with the administrative boundary of Delft with the function `st_read()` from the `sf` package. 
+
+```{r}
+boundary_Delft <- st_read("data/delft-boundary.shp", quiet = TRUE)
```

Comment:
why don't we use the here function? 
e.g. 
boundary_Delft <- st_read(here("data", "delft-boundary.shp"), quiet = TRUE)

Reply:
I had an issue with inline R code from the paragraph following this chunk. The inline code would not run unless I call st_read with a regular path and I don't know why... I'll try to fix it, as `here` should indeed be used consistently throughout the workshop.

Reply:
I tried to knit this Rmd with the here function to HTML. However, even though the here package was activated, I received the following comment. This issue only appears when I try to render the Rmd. Did you have the same issue?
<img width="878" alt="Screenshot 2024-01-18 at 10 03 37" src="https://github.com/carpentries-incubator/r-geospatial-urban/assets/129063142/09804874-6e79-40be-9755-f801b66cec07">

Reply:
here is the line 60: 
```{r}
boundary_Delft <- st_read(here("data", "delft-boundary.shp"), quiet = TRUE)
```

Reply:
Knitting does not always work in workbench lessons. Always build the website to check the output. For some reason I also had an issue with `here()` so I decided to use the path directly.